src/gluonts/model/statsforecast.py

```diff
@@ -0,0 +1,329 @@
+            interval = round(
+                200 * (max(quantile_level, 1 - quantile_level) - 0.5)
+            )
+            intervals.add(interval)
+            side = "hi" if quantile_level > 0.5 else "lo"
+            self.forecast_keys.append(str(quantile_level))
+            self.statsforecast_keys.append(f"{side}-{interval}")
+
+        self.intervals = sorted(list(intervals))
+
+
+class StatsForecastPredictor(RepresentablePredictor):
+    """
+    A predictor type that wraps models from the `statsforecast`_ package.
+
+    Objects of this class are constructed with a type (for example,
+    ``statsforecast.models.AutoARIMA``) and a dictionary of parameters.
+
+    .. _statsforecast: https://github.com/Nixtla/statsforecast
+
+    Parameters
+    ----------
+    model_constructor
+        Model constructor (e.g. ``statsforecast.models.AutoARIMA``).
+    kwargs
+        Dictionary of keyword arguments to be passed for model construction.
+    prediction_length
+        Prediction length for the model to use.
+    quantile_levels
+        Optional list of quantile levels that we want predictions for.
+        By default this is ``None``, giving only the mean predition.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        model_constructor,
+        kwargs: dict,
+        prediction_length: int,
+        quantile_levels: Optional[List[float]] = None,
+    ) -> None:
+        super().__init__(prediction_length=prediction_length)
+        self.model = model_constructor(**kwargs)
+        self.config = ModelConfig(quantile_levels=quantile_levels)
+
+    def predict_item(self, entry: DataEntry) -> QuantileForecast:
+        # TODO use also exogenous features
+        kwargs = {}
+        if self.config.intervals is not None:
+            kwargs["level"] = self.config.intervals
+
+        pred = self.model.forecast(
+            y=entry["target"],
+            h=self.prediction_length,
+            **kwargs,
+        )
+
+        forecast_arrays = [pred[k] for k in self.config.statsforecast_keys]
+
+        return QuantileForecast(
+            forecast_arrays=np.stack(forecast_arrays, axis=0),
+            forecast_keys=self.config.forecast_keys,
+            start_date=forecast_start(entry),
+            item_id=entry.get("item_id"),
+            info=entry.get("info"),
+        )
+
+
+class ADIDAPredictor(StatsForecastPredictor):
+    """
+    A predictor wrapping the ``ADIDA`` model from `statsforecast`_.
+
+    The predictor is constructed with a ``prediction_length`` parameters,
+    plus all additional arguments needed by ``ADIDA``: please refer to
+    the documentation of `statsforecast`_ to know more about its parameters.
+
+    .. _statsforecast: https://github.com/Nixtla/statsforecast
+    """
+
+    @validated()
+    def __init__(
+        self, prediction_length: int, quantile_levels=None, **kwargs
+    ) -> None:
+        super().__init__(
+            model_constructor=ADIDA,
```

Comment:
Can't we just make this a class variable? Then we don't need to have this `__init__` at all.